lighthouse-core/computed/metrics/cumulative-layout-shift.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('../computed-artifact.js');
+const TraceOfTab = require('../trace-of-tab.js');
+const LHError = require('../../lib/lh-error.js');
+
+class CumulativeLayoutShift {
+  /**
+   * @param {LH.Trace} trace
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Artifacts.MetricValue>}
+   */
+  static async compute_(trace, context) {
+    if (!trace) {
```

Comment:
this feels like something we don't do normally :)

I know we do for the base metric class but that's just because those get required in by all the opportunities that indirectly call things in a weird web where something was getting dropped along the way. this feels much less likely to slip through tsc's fingers

Reply:
+1

we also assert DEFAULT_PASS exists if an audit requires `traces`, so even ignoring type checking, this should always exist if it gets this far